### DISCUSSION TITLE
How do I get my patterns to show up in the command palette?

### DISCUSSION THREAD

**Discussion by olivia530** [jglev/obsidian-apply-patterns-plugin#25] [Q&A]
Hey, thanks for making this plugin—it seems to be exactly what I've been looking for. 

One thing I am having trouble with is getting my patterns to show up in the command palette. When I bring up the command palette, search "Apply Patterns," and select one of the three default options, none of the patterns I've created show up. Instead I get basically a blank search bar with no options to select. This means the plugin is functionally unusable as I cannot execute any of my patterns.

Anyone have any clue what could be causing this?

**Comment by olivia530** [comment]
I've figured out a workaround and possibly stumbled across a bug in the process.

At the time I posted this I had only created one pattern so far. However, I just added a second pattern, and this time when I brought up the command palette and selected one of the default "Apply Patterns" options, I could see the *first* pattern I'd added but not the new one. I then added a third test pattern with an arbitrary regex rule. This time when I called up the command palette I could see my first two patterns and not the test pattern.

In further experimentation I've discovered that the test pattern cannot be empty or both it and the pattern prior to it are missing from the command palette.

To illustrate what I'm talking about, here's a screenshot of my three patterns:
<img width="666" alt="screenshot_115" src="https://user-images.githubusercontent.com/90682443/136512496-262df429-e3b4-48ca-a992-fc2d6b245ecb.png">

And here is what the command palette looks like if I call up "Apply Patterns" and select an option:
<img width="702" alt="screenshot_116" src="https://user-images.githubusercontent.com/90682443/136512540-6b7af421-1f3b-46d8-adc0-3ee0809145e9.png">

**Comment by jglev** [comment]
This is too late for a response, but I'm coming back through and clearing out old threads. I *think* that this describes the same behavior that was later fixed with v2.1.2. (See [this thread](https://github.com/jglev/obsidian-apply-patterns-plugin/issues/58).)

I'm sorry for missing this when it was posted; I am very grateful (even posting several years later) for the detailed write-up!
